### ISSUE TITLE
I extend NBGrader to enable the automated assessment of SQL statements

### ISSUE THREAD

**Issue by Tshimangadzo** [jupyter/nbgrader#1031]
I am new to **Nbgrader**, i come up with the problem on how can i extend NBGrader to enable the automated assessment of **SQL statements**

**Comment by jhamrick** [comment]
nbgrader isn't really designed to grade any particular type of
language---even though the default is python, it is meant to work with any
type of Jupyter kernel. As far as I know there isn't a kernel specifically
for SQL, but I believe you can run SQL commands in the python kernel using
the `%%sql` magic. But overall it is up to you as an instructor to design
the tests for checking that the sql commands which were run are correct. So
in other words it's not really nbgrader's responsibility to support a
specific language (like SQL), that's the job of the instructor.

**Comment by perllaghu** [comment]
When you say _assessment of SQL statements_ are you looking to check they're "well formed" statements, or that they produce valid output?

To check they're "well formed", you can use libraries like `sqlparse` 

But _valid_ is a whole different ballgame - see https://www.reddit.com/r/learnpython/comments/72ec9h/sql_validator_in_python/

**Comment by jhamrick** [comment]
I am closing this issue because there hasn't been any activity on it in a while. However, if you (or others) need any more help, feel free to open a new issue!
